admin_guide/install/advanced_install.adoc

```diff
@@ -137,42 +160,202 @@ master.example.com openshift_node_labels="{'region': 'infra', 'zone': 'default'}
 node1.example.com openshift_node_labels="{'region': 'primary', 'zone': 'east'}"
 node2.example.com openshift_node_labels="{'region': 'primary', 'zone': 'west'}"
 ----
-endif::[]
-ifdef::openshift-origin[]
+====
+
+Modify the file to match your environment and specifications, and save it as
+*_/etc/ansible/hosts_*.
+
+[[single-master-multi-etcd-multi-node]]
```

Comment:
@timothysc @adellape One example I came across recently was helping someone achieve a strategy for an HA deployment in OpenStack.

The main issues that were presented were:
- Neutron network using the Nuage plugin didn't provide the feature needed to be able to enable configuring a VIP through pacemaker
- OpenStack doesn't provide a facility for giving a host permission to associate/disassociate a floating IP like you can do with ec2, so configuring pacemaker with a custom script to associate/disassociate a floating IP would require that the user pass actual OpenStack credentials to the host if they are going to have a script to handle migrating the floating IP as part of cluster failover

The solution we came up with was as follows:
- Configure a single master env with HA etcd using openshift-ansible ensuring that the master host is provisioned to use a persistent disk.
- Create a Heat template to provide HA for a master instance based off of the persistent disk used for installing OpenShift.
  - Heat provides the ability to do service level checks as well as system level checks, so if the service is unavailable it will attempt to restart the service up to 3 times, and then will re-provision the host if unsuccessful.

While the solution doesn't provide the same level of HA as using pacemaker, it does provide a higher level of protection than otherwise available in the environment.